### PR TITLE
Kfold validation

### PR DESCRIPTION
**Pull request by diegodimer**
- [x] adicionar kfold do outro trabalho
- [x] adicionar datasets
- [x] kfold funcionar

### CODE REVIEW

File: neuralNetwork.py [hunk 1]
```diff
@@ -168,18 +206,17 @@ def backpropagation(self, test_set, debug=False):
...
                     gr = str(gradients[i])
                     print(re.sub(r"[\[\]]", r"",gr))
-            
+
 
             if not debug:
                 # 3. atualizar cada peso com o valor do gradiente:
-                for i in range(len(self.theta_list)):                  
+                for i in range(len(self.theta_list)):
                     self.theta_list[i] = self.theta_list[i] - self.learning_rate * (gradients[i] + self.reg_factor*self.theta_list[i])
```
Comment (diegodimer) [inline neuralNetwork.py]: image
aqui não devia ter um J?

Comment (ecpaim) [reply neuralNetwork.py]: Essa é uma dúvida que eu tive com os slides:
Screenshot from 2020-11-06 00-31-48
eu acho que essa notação matemática da esquerda nem é válida, ta tirando o gradiente de d[ ]/dtheta? eu _presumo_ que ele esteja tirando o gradiente da função J, então o `gradients[i]` que é calculado no backpropagation é esse gradiente que é usado na atualização. Porque não teria como colocar o gradiente como input na função de J, não faz sentido

Comment (diegodimer) [reply neuralNetwork.py]: tá, é que é delta de J pq é o gradiente da função J

Comment (diegodimer) [reply neuralNetwork.py]: dJ/dtheta no caso
